backend/asmgen.ml

```diff
@@ -448,7 +461,11 @@ let compile_phrases ~ppf_dump ps =
           if !dump_cmm then fprintf ppf_dump "%a@." Printcmm.phrase p;
           match p with
           | Cfunction fd ->
-            compile_fundecl ~ppf_dump ~funcnames fd;
+            let compile_fundec = compile_fundecl ~ppf_dump ~funcnames in
+            let profiled_compile_fundec = match !profile_granularity with
+            | Function_level -> Profile.record ~accumulate:true fd.fun_name.sym_name compile_fundec
+            | _ -> compile_fundec (* Don't profile if -dfunc-level not selected*)
```

Comment:
```suggestion
            | File_level -> compile_fundec (* Don't profile if -dfunc-level not selected*)
```
(It is often better to avoid catch-all pattern,
to be notified by the compiler when new
constructors are added to a type.)

Reply:
In this case, would it be okay since this part of the code should only be executed if the granularity is `Function_level`? So, even if new granularities are added theoretically this should not change (unless we add a more fine-grained granularity than `Function_level` which seems unlikely).

Reply:
Generally, we prefer to have to perform
an obvious code update rather than take
the risk of missing an update. Warning 4
is enabled in most files in the backend,
but is unfortunately still disabled in
`Asmgen`.